pages/en/lb3/Polymorphic-relations.md

```diff
@@ -147,80 +289,77 @@ Then here's an example of a polymorphic hasManyThrough relation:
     "pictures": {
       "type": "hasMany",
       "model": "Picture",
-      "polymorphic": {
-        "foreignKey": "imageableId",
-        "discriminator": "imageableType"
-      },
       "through": "ImageLink",
-      "keyThrough": "authorId"
+      "polymorphic": "imageable"
     }
   },
 ...
 }
 ```
 
-Equivalently, in JavaScript:
+OPTIONAL in model Picture.
 
-{% include code-caption.html content="/common/models/Author.js" %}
-```javascript
-Author.hasMany(Picture, {
-  as: 'pictures',
-  polymorphic: {
-    foreignKey: 'imageableId',
-    discriminator: 'imageableType'
-  },
-  through: ImageLink,
-  keyThrough: 'authorId'
-});
-```
-
-## BelongsTo polymorphic relations
-
-Because you define the related model dynamically, you cannot declare it up front.
-So instead of passing in the related model (name), you specify the name of the polymorphic relation.
-
-{% include code-caption.html content="common/models/picture.json" %}
+{% include code-caption.html content="/common/models/Picture.json" %}
 ```javascript
+// Optional, define invert hasMany relation in Picture
 {
   "name": "Picture",
   "base": "PersistedModel",
   ...
   "relations": {
-    "author": {
-      "type": "belongsTo",
+    "authors": {
+      "type": "hasMany",
       "model": "Author",
-      "polymorphic": {
-        "foreignKey": "imageableId",
-        "discriminator": "imageableType"
-      }
+      "through": "ImageLink",
+      "invert": true,
+      "polymorphic": "imageable"
+    },
+    "readers": {
+      "type": "hasMany",
+      "model": "Reader",
+      "through": "ImageLink",
+      "invert": true,
```

Comment:
@b-admike I explained it in https://github.com/strongloop/loopback.io/pull/381#issue-228865439

My current understanding is the invert hasMany relation is optional(I think this is explained in doc, I also verified it), but regarding why do we had the optional relation in old doc(and juggler test), I could not say for sure, probably because by doing that you can have the api on model `Picture`